Makefile

```diff
@@ -0,0 +1,49 @@
+.PHONY: check doc faktory faktory/* test test/*
```

Comment:
I don't know if `*` actually works here. Usually, Make is very particular about what `*` means, and I don't think it can wildcard rules this way when they aren't targeting pre-existing files.

Reply:
I tested it with `README.md` example where putting `RE*` to the list of phony targets will deem `README.md` always stale

Reply:
putting as little as 1letter with a wildcard `R*` to the list will be enough to make it recreate the recipe ...
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/ffb1eaea-3a41-4ec9-a43d-55ee722ece16)


while putting `*` will cover all the commands:
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/8a565195-9a95-442b-98ba-0d855f1b4adc)

Reply:
Thanks for checking!

Reply:
Looked through some repos. 

Indeed, this is the guidelines  they follow in the [linux kernel](https://github.com/torvalds/linux/blob/master/Makefile#L1957-L1960) repo, though  with a little adaptation - via a variable collecting the phony targets across the file and announcing them phony at the very bottom. When the file gets long, it makes it more explicit, definitely. 

In [https://github.com/redis/redis/blob/unstable/Makefile] the file is a few lines of code, and I am not sure if  they meant to put the `.PHONY` declaration at the very bottom. In [RabbitMQ](https://github.com/rabbitmq/rabbitmq-server/blob/main/Makefile#L374-L408) they declare phony target per section, which is almost the same as per target due to how they visually separate those section.

Updated according to the guidelines and checked with command:
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/bee31138-5230-4f59-8fc4-a7c00566db2f)

`test/e2e` is _now_ really recognized as a phony target:
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/f0d75aa2-d51a-4aa5-954b-abb63c275db1)

Reply:
Excellent, thanks!